### PR TITLE
API-7819 - Replace scalaj with sttp

### PR DESCRIPTION
**Pull request by AndySpaven**

### CODE REVIEW

File: it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala [hunk 1]
```diff
@@ -25,18 +25,32 @@ import com.github.tomakehurst.wiremock.client.WireMock
...
+import sttp.client3.{Request, Response, SimpleHttpClient}
 
 import play.api.Application
 import play.api.inject.guice.GuiceApplicationBuilder
 
 import uk.gov.hmrc.testuser.helpers.stubs.AuthLoginApiStub
 import uk.gov.hmrc.testuser.repository.TestUserRepository
 
-trait BaseSpec extends AnyFeatureSpec with BeforeAndAfterAll with BeforeAndAfterEach with Matchers with GuiceOneServerPerSuite
+trait BaseFeatureSpec
```
Comment (johnsgp) [inline it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala:38]: Please rename the file too

File: it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala [hunk 2]
```diff
@@ -25,18 +25,32 @@ import com.github.tomakehurst.wiremock.client.WireMock
...
+    with BeforeAndAfterAll
+    with BeforeAndAfterEach
+    with Matchers
+    with GuiceOneServerPerSuite
+    with EitherValues
     with GivenWhenThen {
 
+  def http(request: => Request[Either[String, String], Any]): Response[Either[String, String]] = {
+    val httpClient = SimpleHttpClient()
+    val response   = httpClient.send(request.followRedirects(false))
```
Comment (johnsgp) [inline it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala:49]: Do we need `.followRedirects(false)`?

File: it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala (hunk 1 above)
Comment (AndySpaven) [reply it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala:38]: Refactor failure :-)

File: it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala (hunk 2 above)
Comment (AndySpaven) [reply it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala:49]: No - that's typically only useful when things want to test where the redirect would send you.

Comment (johnsgp) [reply it/test/uk/gov/hmrc/testuser/helpers/BaseSpec.scala:49]: I meant, should it be removed from the code, as it's purpose might not be clear.

**Review by johnsgp — APPROVED** [review]

**Pull request merged**